spacy/training/loggers.py

```diff
@@ -23,13 +26,39 @@ def setup_table(
     return final_cols, final_widths, ["r" for _ in final_widths]
 
 
-@registry.loggers("spacy.ConsoleLogger.v1")
-def console_logger(progress_bar: bool = False):
+@registry.loggers("spacy.ConsoleLogger.v2")
+def console_logger(
+    progress_bar: bool = False,
+    console_output: bool = True,
+    output_file: Optional[Union[str, Path]] = None,  # type: ignore
+):
+    """The ConsoleLogger.v2 prints out training logs in the console and/or saves them to a jsonl file.
+    progress_bar (bool): Whether the logger should print the progress bar.
+    console_output (bool): Whether the logger should print the logs on the console.
+    output_file (Optional[Union[str, Path]]): The file to save the training logs to.
+    """
+    _log_exist = False
+    if output_file:
+        output_file = util.ensure_path(output_file)  # type: ignore
+        if output_file.exists():  # type: ignore
+            _log_exist = True
+        if not output_file.parents[0].exists():  # type: ignore
+            output_file.parents[0].mkdir(parents=True)  # type: ignore
+
     def setup_printer(
         nlp: "Language", stdout: IO = sys.stdout, stderr: IO = sys.stderr
     ) -> Tuple[Callable[[Optional[Dict[str, Any]]], None], Callable[[], None]]:
         write = lambda text: print(text, file=stdout, flush=True)
         msg = Printer(no_print=True)
+
+        nonlocal output_file
+
+        if _log_exist:
+            write(msg.warn(f"The log file {output_file} already exists."))
+            output_file = None
+        elif output_file:
+            write(msg.info(f"Saving logs to {output_file}"))
```

Comment:
No `finally` needed because this is already handled in the training loop, this is what I would expect to work (aside from the types, which continue to be a hassle for optional paths):

```diff
diff --git a/spacy/training/loggers.py b/spacy/training/loggers.py
index 13999ceca..e4d81d257 100644
--- a/spacy/training/loggers.py
+++ b/spacy/training/loggers.py
@@ -52,12 +52,13 @@ def console_logger(
         msg = Printer(no_print=True)

         nonlocal output_file
-
+        output_fileh = None
         if _log_exist:
             write(msg.warn(f"The log file {output_file} already exists."))
             output_file = None
         elif output_file:
             write(msg.info(f"Saving logs to {output_file}"))
+            output_fileh = open(output_file, "w")

         # ensure that only trainable components are logged
         logged_pipes = [
@@ -124,9 +125,8 @@ def console_logger(
                 "scores": log_scores,
                 "score": float(info["score"]),
             }
-            if output_file:
-                with (output_file).open("a") as file_:  # type: ignore
-                    file_.write(srsly.json_dumps(log_data) + "\n")
+            if output_fileh:
+                output_fileh.write(srsly.json_dumps(log_data) + "\n")

             if progress is not None:
                 progress.close()
@@ -144,7 +144,8 @@ def console_logger(
                     progress.set_description(f"Epoch {info['epoch']+1}")

         def finalize() -> None:
-            pass
+            output_fileh.close()
+

         return log_step, finalize

```